src/plugins/flexibleLayout/components/flexibleLayout.vue

```diff
@@ -0,0 +1,676 @@
+        ResizeHandle,
+        DropHint
+    },
+    data() {
+        let containers = this.domainObject.configuration.containers,
+            rowsLayout = this.domainObject.configuration.rowsLayout;
+
+        if (!containers.length) {
+            containers = [new Container(50), new Container(50)];
+        }
+
+        return {
+            containers: containers,
+            dragFrom: [],
+            isEditing: false,
+            isDragging: false,
+            isContainerDragging: false,
+            rowsLayout: rowsLayout,
+            maxMoveSize: 0
+        }
+    },
+    computed: {
+        layoutDirectionStr() {
+            if (this.rowsLayout) {
+                return 'Rows'
+            } else {
+                return 'Columns'
+            }
+        }
+    },
+    methods: {
+        areAllContainersEmpty() {
+            return !!!this.containers.filter(container => container.frames.length > 1).length;
+        },
+        addContainer() {
+            let newSize = 100/(this.containers.length+1);
+
+            let container = new Container(newSize)
+
+            this.recalculateContainerSize(newSize);
+
+            this.containers.push(container);
+        },
+        recalculateContainerSize(newSize) {
+            this.containers.forEach((container) => {
+                container.width = newSize;
+            });
+        },
+        recalculateNewFrameSize(multFactor, framesArray){
+            framesArray.forEach((frame, index) => {
+                if (index === 0) {
+                    return;
+                }
+                if (framesArray.length === 2) {
+                    frame.height = 100;
+                } else {
+                    let frameSize = frame.height
+                    frame.height = this.snapToPercentage(multFactor * frameSize);
+                }
+            });
+        },
+        recalculateOldFrameSize(framesArray) {
+            let totalRemainingSum = framesArray.map((frame,i) => {
+                if (i !== 0) {
+                    return frame.height
+                } else {
+                    return 0;
+                }
+            }).reduce((a, c) => a + c);
+
+            framesArray.forEach((frame, index) => {
+
+                if (index === 0) {
+                    return;
+                }
+                if (framesArray.length === 2) {
+
+                    frame.height = 100;
+                } else {
+
+                    let newSize = frame.height + ((frame.height / totalRemainingSum) * (100 - totalRemainingSum));
+                    frame.height = this.snapToPercentage(newSize);
+                }
+            });
+        },
+        addFrame(frame, index) {
+            this.containers[index].addFrame(frame);
+        },
+        frameDragFromHandler(containerIndex, frameIndex) {
+            this.dragFrom = [containerIndex, frameIndex];
+        },
+        frameDropToHandler(containerIndex, frameIndex, frameObject) {
```

Comment:
We might reconsider the sizing logic here. When I tested this out by dragging and dropping three elements into a column, I expected them to get equal height (33, 33, 33). But instead they got (45, 22, 33) - see image.

![screen shot 2018-11-08 at 1 21 03 pm](https://user-images.githubusercontent.com/4215777/48228670-b3ecc300-e35a-11e8-9659-505e6e068d58.png)

Looking at the logic here, it seems that a newly dropped object gets 50%, which is then used to calculate a rescaling factor for the rest of the frames.

I think there's an approach that results in more predictable sizing here -

1. When a new object is dropped in a row or column, add it to the frame. Rather than using 50 as its initial size,  calculate its initial size using -
```js
let initialSize = 100 / frames.length
```
2. Calculate the new total. ie 
```js
let totalSize = frames.reduce((total, frame) => total + frame.height, 0);
```
3. Calculate a scaling factor 
```js
let scalingFactor = 100 / totalSize
```
4. Resize all frames (including the new one) using the scaling factor
```js
frames.forEach(frame => frame.height = frame.height * scalingFactor);
```

Reply:
Above algorithm has a bug, stand by...